core/embed/rust/src/micropython/runtime.rs

```diff
@@ -0,0 +1,10 @@
+use cstr_core::CStr;
+
+use super::ffi;
+
+pub fn raise_value_error(msg: &'static CStr) -> ! {
+    unsafe {
+        ffi::mp_raise_ValueError(msg.as_ptr());
```

Comment:
As previously discussed, I'm wondering whether exceptions raised in python code called from rust cannot bite us in some way. IIUC this means we cannot rely on destructors (`Drop::drop()`) always being executed. It shouldn't lead to memory leaks since heap is managed by micropython GC, but we have to keep this in mind & review all dependencies whether they rely on Drop for correctness.

Is it feasible to install exception handler (`nlr_push()`) every time we call rust->python and either panic when caught (imho reasonable for allocation failures), or choose to propagate the exception using `Result`s to the underlying caller?

Reply:
It should be feasible, yes. There is some performance hit, but if we're actually calling Python it won't be relatively big, and we could replace the cases of C code that only sometimes call Python or raise exceptions themselves, like most of the conversion methods, with our non-raising variants.

More about leaking here: https://doc.rust-lang.org/nomicon/leaking.html